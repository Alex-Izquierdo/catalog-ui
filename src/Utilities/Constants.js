export const DEFAULT_PAGE_SIZE = 50;

const topoApiHost = process.env.TOPO_API_HOST || '';
const topoApiPort = process.env.TOPO_API_PORT || '443';
const topoBasePath = process.env.TOPO_BASE_PATH;
const topoApiVersion = process.env.TOPO_API_VERSION || '/api/v0.0';
export const SOURCES_API_BASE = `https://${topoApiHost}:${topoApiPort}${topoBasePath}${topoApiVersion.replace(/\/+$/, '')}`;

const apiHost = process.env.API_HOST || 'localhost';
const apiPort = process.env.API_PORT || '5000';
const basePath = process.env.BASE_PATH || '';
const apiVersion = process.env.API_VERSION || '/api/v0.0';
export const SERVICE_PORTAL_API_BASE = `https://${apiHost}:${apiPort}${basePath}${apiVersion.replace(/\/+$/, '')}`;
