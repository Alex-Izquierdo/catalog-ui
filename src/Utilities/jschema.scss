@import "../App.scss";
@import '../../node_modules/bootstrap/dist/css/bootstrap.css';


.rjsf {
  border: 10px;
  padding-top: 20px;
  padding-left: 20px;
  padding-right: 20px;
  max-width: 500px;

  font-size: small;
}
.form-group {
  margin-left: 15px;
  margin-bottom: 15px;
}


.hidden {
  display: none;
}



