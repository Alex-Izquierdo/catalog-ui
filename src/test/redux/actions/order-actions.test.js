import configureStore from 'redux-mock-store' ;
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { notificationsMiddleware, ADD_NOTIFICATION } from '@red-hat-insights/insights-frontend-components/components/Notifications';
import { SERVICE_PORTAL_API_BASE } from '../../../Utilities/Constants';
import {
  fetchServicePlans,
  fetchOrderList,
  updateServiceData,
  setSelectedPlan,
  sendSubmitOrder
} from '../../../redux/Actions/OrderActions';
import {
  FETCH_SERVICE_PLANS,
  LIST_ORDERS,
  UPDATE_SERVICE_DATA,
  SET_SELECTED_PLAN,
  SUBMIT_SERVICE_ORDER
} from '../../../redux/ActionTypes';

describe('Order actions', () => {
  const middlewares = [ thunk, promiseMiddleware(), notificationsMiddleware() ];
  let mockStore;

  beforeEach(() => {
    mockStore = configureStore(middlewares);
  });

  it('should dispatch correct actions after fetching service plans', () => {
    const store = mockStore({});
    apiClientMock.get(`${SERVICE_PORTAL_API_BASE}/portfolio_items/1/service_plans`, mockOnce({
      body: [ 'Foo' ]
    }));
    const expectedActions = [{
      type: `${FETCH_SERVICE_PLANS}_PENDING`
    }, expect.objectContaining({
      type: `${FETCH_SERVICE_PLANS}_FULFILLED`
    }) ];

    return store.dispatch(fetchServicePlans(1)).then(() => {
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  it('should dispatch correct actions after fetching service plans fails', () => {
    const store = mockStore({});
    apiClientMock.get(`${SERVICE_PORTAL_API_BASE}/portfolio_items/1/service_plans`, mockOnce({
      status: 500
    }));
    const expectedActions = [{
      type: `${FETCH_SERVICE_PLANS}_PENDING`
    }, expect.objectContaining({
      type: ADD_NOTIFICATION,
      payload: expect.objectContaining({
        variant: 'danger'
      })
    }), expect.objectContaining({
      type: `${FETCH_SERVICE_PLANS}_REJECTED`
    }) ];

    return store.dispatch(fetchServicePlans(1)).catch(() => {
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  it('should dispatch correct actions after fetching orders', () => {
    const store = mockStore({});
    apiClientMock.get(`${SERVICE_PORTAL_API_BASE}/orders`, mockOnce({
      body: [ 'Foo' ]
    }));
    const expectedActions = [{
      type: `${LIST_ORDERS}_PENDING`
    }, expect.objectContaining({
      type: `${LIST_ORDERS}_FULFILLED`
    }) ];

    return store.dispatch(fetchOrderList()).then(() => {
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  it('should dispatch correct actions after fetching orders fails', () => {
    const store = mockStore({});
    apiClientMock.get(`${SERVICE_PORTAL_API_BASE}/orders`, mockOnce({
      status: 500
    }));
    const expectedActions = [{
      type: `${LIST_ORDERS}_PENDING`
    }, expect.objectContaining({
      type: ADD_NOTIFICATION,
      payload: expect.objectContaining({
        variant: 'danger'
      })
    }), expect.objectContaining({
      type: `${LIST_ORDERS}_REJECTED`
    }) ];

    return store.dispatch(fetchOrderList()).catch(() => {
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  it('should dispatch correct actions after calling update service data', () => {
    const store = mockStore({});
    const expectedActions = [{
      type: UPDATE_SERVICE_DATA,
      payload: { serviceData: { foo: 'bar' }}
    }];

    store.dispatch(updateServiceData({ foo: 'bar' }));
    expect(store.getActions()).toEqual(expectedActions);
  });

  it('should dispatch correct actions after calling set selected plan', () => {
    const store = mockStore({});
    const expectedActions = [{
      type: SET_SELECTED_PLAN,
      payload: { foo: 'bar' }
    }];

    store.dispatch(setSelectedPlan({ foo: 'bar' }));
    expect(store.getActions()).toEqual(expectedActions);
  });

  it('should dispatch correct actions after submitting order', () => {
    const store = mockStore({});
    apiClientMock.post(`${SERVICE_PORTAL_API_BASE}/orders`, mockOnce({
      body: { id: 123 }
    }));
    apiClientMock.post(`${SERVICE_PORTAL_API_BASE}/orders/123/order_items`, mockOnce((req, res) => {
      expect(JSON.parse(req.body())).toEqual({
        count: 1,
        provider_control_parameters: { namespace: 'default' },
        portfolio_item_id: 'Foo',
        service_plan_ref: 'Bar',
        service_parameters: { bax: 'quxx' }
      });
      return res.status(200);
    }));
    apiClientMock.post(`${SERVICE_PORTAL_API_BASE}/orders/123/submit_order`, mockOnce({
      status: 200
    }));
    const expectedActions = [{
      type: `${SUBMIT_SERVICE_ORDER}_PENDING`
    }, expect.objectContaining({
      type: `${SUBMIT_SERVICE_ORDER}_FULFILLED`
    }) ];

    return store.dispatch(sendSubmitOrder({
      portfolio_item_id: 'Foo',
      service_plan_ref: 'Bar',
      service_parameters: { bax: 'quxx', providerControlParameters: { namespace: 'default' }}
    })).then(() => {
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  it('should dispatch correct actions after submitting order fails to get new order', () => {
    const store = mockStore({});
    apiClientMock.post(`${SERVICE_PORTAL_API_BASE}/orders`, mockOnce({
      status: 500
    }));
    const expectedActions = [{
      type: `${SUBMIT_SERVICE_ORDER}_PENDING`
    }, expect.objectContaining({
      type: ADD_NOTIFICATION,
      payload: expect.objectContaining({
        variant: 'danger'
      })
    }), expect.objectContaining({
      type: `${SUBMIT_SERVICE_ORDER}_REJECTED`
    }) ];

    return store.dispatch(sendSubmitOrder({
      portfolio_item_id: 'Foo',
      service_plan_ref: 'Bar',
      service_parameters: 'Quxx'
    })).catch(() => {
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  it('should dispatch correct actions after submitting order fails to add to new order', () => {
    const store = mockStore({});
    apiClientMock.post(`${SERVICE_PORTAL_API_BASE}/orders`, mockOnce({
      body: { id: 123 }
    }));

    apiClientMock.post(`${SERVICE_PORTAL_API_BASE}/orders/123/items`, mockOnce({
      status: 500
    }));

    const expectedActions = [{
      type: `${SUBMIT_SERVICE_ORDER}_PENDING`
    }, expect.objectContaining({
      type: ADD_NOTIFICATION,
      payload: expect.objectContaining({
        variant: 'danger'
      })
    }), expect.objectContaining({
      type: `${SUBMIT_SERVICE_ORDER}_REJECTED`
    }) ];

    return store.dispatch(sendSubmitOrder({})).catch(() => {
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  it('should dispatch correct actions after submitting order fails to submit order', () => {
    const store = mockStore({});
    apiClientMock.post(`${SERVICE_PORTAL_API_BASE}/orders`, mockOnce({
      body: { id: 123 }
    }));

    apiClientMock.post(`${SERVICE_PORTAL_API_BASE}/orders/123/items`, mockOnce({
      status: 200
    }));
    apiClientMock.post(`${SERVICE_PORTAL_API_BASE}/orders/123`, mockOnce({
      status: 500
    }));

    const expectedActions = [{
      type: `${SUBMIT_SERVICE_ORDER}_PENDING`
    }, expect.objectContaining({
      type: ADD_NOTIFICATION,
      payload: expect.objectContaining({
        variant: 'danger'
      })
    }), expect.objectContaining({
      type: `${SUBMIT_SERVICE_ORDER}_REJECTED`
    }) ];

    return store.dispatch(sendSubmitOrder({})).catch(() => {
      expect(store.getActions()).toEqual(expectedActions);
    });
  });
});
