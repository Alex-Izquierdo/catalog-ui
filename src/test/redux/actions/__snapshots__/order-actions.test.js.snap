// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order actions should dispatch correct actions after calling getLinkedOrders and create correct data structure 1`] = `
Array [
  Object {
    "type": "FETCH_LINKED_ORDERS_PENDING",
  },
  Object {
    "payload": Object {
      "current": Array [
        Object {
          "id": "1",
          "name": "super-order",
          "orderItems": Array [
            Object {
              "name": "super-order-item",
              "order_id": "1",
            },
          ],
          "requests": Array [
            Object {
              "content": Object {
                "order_id": "1",
              },
              "name": "super-order-request",
            },
          ],
        },
        Object {
          "id": "2",
          "name": "sad-lonely-order",
          "orderItems": Array [],
          "requests": Array [],
        },
      ],
      "past": Array [
        Object {
          "id": "3",
          "name": "completed-order",
          "orderItems": Array [],
          "requests": Array [],
          "state": "Completed",
        },
        Object {
          "id": "4",
          "name": "failed-order",
          "orderItems": Array [],
          "requests": Array [],
          "state": "Failed",
        },
      ],
    },
    "type": "FETCH_LINKED_ORDERS_FULFILLED",
  },
]
`;
